src/herring/tl_helpers.hpp

```diff
@@ -151,10 +153,25 @@ auto convert_tree(treelite::Tree<tl_threshold_t, tl_output_t> const& tl_tree, bo
           throw unconvertible_model_exception{"Unsupported comparison operator"};
         }
       } else {
-        throw unconvertible_model_exception{"Categorical nodes not yet implemented"};
+        if (tl_tree.CategoriesListRightChild(cur_node_id)) {
+          hot_child = left_id;
+          distant_child = right_id;
+        } else {
+          hot_child = right_id;
+          distant_child = left_id;
+        }
+        auto tl_categories = tl_tree.MatchingCategories(cur_node_id);
+        auto constexpr max_category = typename tree_t::node_type::category_set_type{}.size();
```

Comment:
Is this used?

Reply:
Oh darn! Thank you for catching that. It's not, but it should be. Just pushed an update.